test/functions.js

```diff
@@ -694,6 +694,9 @@
       assert.deepEqual(_.toArray(cb(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)), _.range(1, 11));
     });
 
+    var deepProperty = _.iteratee(['a', 'b']);
+    assert.equal(deepProperty({a: {b: 2}}), 2, 'treats an array as a deep property accessor');
+
```

Comment:
Compat note:
Lodash treats an array as shorthand for `matchesProperty`.

The `matchesProperty` shorthand is `[propName,propValue]` and as fast path for
the common case of `{propName:propValue}`
<img src="https://cloud.githubusercontent.com/assets/4303/14334216/bdac3a34-fc07-11e5-896a-976d6967eda0.png" height=70% width=70%>

Not a biggie but something to be aware of.